### ISSUE TITLE
Concept of Head in A Neural Network

### ISSUE THREAD

**Issue by curimeowcat** [ultralytics/yolov5#7107]
Dear ultralytics team, thanks for this great github repo!

I am trying to understand the concept of "head" in https://github.com/ultralytics/yolov5/blob/ecc2c7ba73e71211b192cba69e255afad92de67a/models/yolov5x.yaml#L28, it would be great if you can explain it.

Specifically, 
(1) How is it different from simply an output from a neural network?
(2) Does each head must have its corresponding labels so that we can compute losses? E.g. we I add a head to output class labels, I can use cross-entropy to train these addition classifier layers.
(3) If I add an additional head, it is simply just some additional layers on top of previous layers?

Thanks in advance!

**Comment by glenn-jocher** [comment]
@curimeowcat 'head' is just one of those names some people like to call different parts of a neural network. The 'head' is usually the last part of a model where the output layers are. If you were so inclined you might have multiple heads for different tasks that exploit a common feature set from earlier in the network.

**Comment by diangu001** [comment]
> @curimeowcat 'head' is just one of those names some people like to call different parts of a neural network. The 'head' is usually the last part of a model where the output layers are. If you were so inclined you might have multiple heads for different tasks that exploit a common feature set from earlier in the network.



> @curimeowcat 'head' is just one of those names some people like to call different parts of a neural network. The 'head' is usually the last part of a model where the output layers are. If you were so inclined you might have multiple heads for different tasks that exploit a common feature set from earlier in the network.

Dear ultralytics team, thanks for this great github repo!
Have you tried replacing YOLO Coupled Head with YOLO Decoupled Head(YOLOX).
